src/multi_processing/coordinator.py

```diff
@@ -0,0 +1,185 @@
+        knowing once they're finished.
+
+    Methods
+    -------
+    create_jobs(domain_obj, quantity, job_size)
+        Populate the generation job queue with jobs
+
+    start_generator(obj_class, pool_size)
+        Begin the generation coordinator as a subprocess, with job pool size
+
+    start_writer(pool_size)
+        Begin the writing coordinator as a subprocess, with job pool size
+
+    get_generation_coordinator()
+        Return the generation coordinator
+
+    get_write_coordinator()
+        Return the write coordinator
+
+    await_termination()
+        Wait for generation & write coordinators to terminate
+    """
+
+    def __init__(self, max_file_size, file_builder):
+        """Create Job Queues for both generation and writing to file, as well
+        as instantiating the coordinators for both.
+
+        Parameters
+        ----------
+        max_file_size : int
+            The maximum number of records to write to each file
+        file_builder : File_Builder
+            Instantiated and pre-configured file builder to write files of
+            the necessary format.
+        """
+
+        queue_manager = Manager()
+        self.__generation_job_queue = queue_manager.Queue()
+        self.__write_job_queue = queue_manager.Queue()
+
+        self.__generation_coordinator = Generator(
+            self.__generation_job_queue,
+            self.__write_job_queue
+        )
+
+        self.__write_coordinator = Writer(
+            self.__write_job_queue,
+            max_file_size,
+            file_builder
+        )
+
+        self.processes = []
+
+    def create_jobs(self, domain_obj, quantity, job_size):
+        """Populate the generation queue with jobs.
+
+        Continually places jobs into the queue, counting down quantity until
+        it's value is 0. Where quantity larger than job size, a job is spawned
+        of job size, and quantity decremented that amount. Where quantity less
+        than job size, a job is spawned of size quantity.
+
+        A job is a 3-element dictionary. Domain object name provided for the
+        purpose of keeping track. Amount and Start_ID are arguments for the
+        generate call. Amount informs as to the number of records to generate.
+        Start_ID keeps track of the batch of IDs the job will be generating
+        in the case of sequentially ID'd domain objects.
+
+        Finally, a termination flag is added to the queue. This informs the
+        coordinator to stop awaiting instruction once read, causing it to
+        terminate once the currently-running jobs have ceased.
+
+        Parameters
+        ----------
+        domain_obj : Generatable
+            Domain object name to be generated
+        quantity : int
+            Number of records to be generated for domain_obj
+        job_size : int
+            Number of records to be generated by each generating process
+        """
+        start_id = 0
+        quantity = int(quantity)
```

Comment:
can we not use 'quanity' since it is already used as in positions. record_count or row_count would be better I think